app/frontend/app/components/group-box.js

```diff
@@ -0,0 +1,51 @@
+import Ember from 'ember';
+/* global Messenger */
+
+export default Ember.Component.extend({
+
+  userIsMemberOfGroup: function(){
+    var users = this.get('group.members').map(function(member){
+      return member.get('user.id');
+    });
+
+    return users.contains(this.get('currentUser.id'));
+  }.property('currentUser', 'group.members.@each'),
+
+  userMembershipIsPending: function(){
+    if(!this.get('userIsMemberOfGroup')) { return false; }
+
+    var user = this.get('group.members').filter((member) => {
+      return member.get('user.id') === this.get('currentUser.id');
+    });
+
+    if(user.length !== 1) { return false; }
+    return user[0].get('pending');
+  }.property('userIsMemberOfGroup'),
+
+
+  actions: {
+    joinGroup: function(){
+      var member = this.get('targetObject.store').createRecord('group-member', {
```

Comment:
Okay, makes sense. I wonder if it would be a good idea to inject the store into components though, would make this a lot cleaner.